lambdas/build-cri-oauth-request/src/main/java/uk/gov/di/ipv/core/buildcrioauthrequest/BuildCriOauthRequestHandler.java

```diff
@@ -275,14 +291,31 @@ private JWEObject signEncryptJar(
             String oauthState,
             String govukSigninJourneyId,
             String criId)
-            throws HttpResponseExceptionWithErrorBody, ParseException, JOSEException {
+            throws HttpResponseExceptionWithErrorBody, ParseException, JOSEException,
+                    UnknownEvidenceTypeException {
+
+        List<String> credentials = userIdentityService.getUserIssuedCredentials(userId);
+
         SharedClaimsResponse sharedClaimsResponse =
-                getSharedAttributesForUser(ipvSessionItem, userId, criId);
+                getSharedAttributesForUser(ipvSessionItem, criId, credentials);
         EvidenceRequest evidenceRequest = null;
+
+        List<SignedJWT> signedJWTS = gpg45ProfileEvaluator.parseCredentials(credentials);
+        Gpg45Scores gpg45Scores = gpg45ProfileEvaluator.buildScore(signedJWTS);
+        List<RequiredGpg45ScoresDto> requiredGpg45Scores =
+                ACCEPTED_PROFILES.stream()
+                        .map(
+                                profile ->
+                                        new RequiredGpg45ScoresDto(
+                                                profile,
+                                                Gpg45ScoresDto.fromGpg45Scores(
+                                                        gpg45Scores.calculateRequiredScores(
+                                                                profile))))
+                        .collect(Collectors.toList());
```

Comment:
Do you think we should put this inside the `if (criId.equals(F2F_CRI))` to avoid calculating it if we don't need it?

Reply:
We could also pull this out into a separate method, which might make it a bit tidier

Reply:
done